tests/snappi_tests/utils/metrics.py

```diff
@@ -0,0 +1,116 @@
+METRIC_LABEL_DEVICE_ID: Final[str] = "device.id"
+METRIC_LABEL_DEVICE_PORT_ID: Final[str] = "device.port.id"
+METRIC_LABEL_DEVICE_PSU_ID: Final[str] = "device.psu.id"
+METRIC_LABEL_DEVICE_QUEUE_ID: Final[str] = "device.queue.id"
+METRIC_LABEL_DEVICE_SENSOR_ID: Final[str] = "device.sensor.id"
+
+
+class PeriodicMetricsReporter:
+    def __init__(self, resource_labels: Dict[str, str]):
+        # Will be replaced with a real initializer such as OpenTelemetry
+        self.resource_labels = deepcopy(resource_labels)
+        self.metrics = []
+
+    def stash_record(self, new_metric: 'Metric', labels: Dict[str, str], value: Union[int, str, float]):
+        # add a new periodic metric
+        copied_labels = deepcopy(labels)
+        self.metrics.append({"labels": copied_labels, "value": value})
+
+    def report(self, timestamp = time.time_ns()):
+        """
+        Report metrics at a given timestamp.
+        The input timestamp must be UNIX Epoch time in nanoseconds since 00:00:00 UTC on 1 January 1970
+
+        # save the metrics in a local variable and release the metrics in the object
+        stashed_metrics = self.metrics
+        self.metrics = []
+
+        """
+        pass
+
+
+class FinalMetricsReporter:
+    def __init__(self, resource_labels: Dict[str, str]):
+        # Will be replaced with a real initializer such as Kusto
+        self.resource_labels = deepcopy(resource_labels)
+        self.metrics = []
+
+    def stash_record(self, new_metric: 'Metric', labels: Dict[str, str], value: Union[int, str, float]):
+        # add a new final metric
+        copied_labels = deepcopy(labels)
+        self.metrics.append({"labels": copied_labels, "value": value})
+
+    def report(self, timestamp = time.time_ns()):
+        """
+        Report metrics at a given timestamp.
+        The input timestamp must be UNIX Epoch time in nanoseconds since 00:00:00 UTC on 1 January 1970
+
+        # save the metrics in a local variable and release the metrics in the object
+        stashed_metrics = self.metrics
+        self.metrics = []
+
+        """
+        pass
+
+
+class Metric:
+    def __init__(self,
+                 name: str,
+                 description: str,
+                 unit: str,
+                 reporter: PeriodicMetricsReporter):
+        """
+        Args:
+            name (str): metric name (e.g., psu power, sensor temperature, port stats, etc.)
+            description (str): brief description of the metric
+            unit (str): metric unit (e.g., seconds, bytes)
+            reporter (PeriodicMetricsReporter): object of PeriodicMetricsReporter
+        """
+        self.name = name
+        self.description = description
+        self.unit = unit
+        self.reporter = reporter
+
+    def __repr__(self):
+        return (f"Metric(name={self.name!r}, "
+                f"description={self.description!r}, "
+                f"unit={self.unit!r}, "
+                f"reporter=repr(self.reporter))")
```

Comment:
will repr(reporter) output anything meaningful?

Reply:
Returning the associated reporter's name makes a better sense. Do you want to change to that?

Reply:
yea, as long as it is readable. returning an address back is not going to help the users.